src/test/java/uk/gov/hmcts/reform/hmc/config/MessageProcessorTest.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.hmc.config;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder;
+import uk.gov.hmcts.reform.hmc.ApplicationParams;
+import uk.gov.hmcts.reform.hmc.client.futurehearing.ActiveDirectoryApiClient;
+import uk.gov.hmcts.reform.hmc.client.futurehearing.AuthenticationResponse;
+import uk.gov.hmcts.reform.hmc.client.futurehearing.HearingManagementInterfaceApiClient;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.BDDMockito.given;
+
+class MessageProcessorTest {
+
+    private static final ObjectMapper OBJECT_MAPPER = new Jackson2ObjectMapperBuilder()
+        .modules(new Jdk8Module())
+        .build();
+    private String requestString;
+    private MessageProcessor messageProcessor;
+
+    @Mock
+    private ApplicationParams applicationParams;
+
+    @Mock
+    private ActiveDirectoryApiClient activeDirectoryApiClient;
+
+    @Mock
+    private HearingManagementInterfaceApiClient hmiClient;
+
+    @BeforeEach
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
+        messageProcessor = new MessageProcessor(applicationParams, activeDirectoryApiClient, hmiClient);
+        requestString = "grant_type=GRANT_TYPE&client_id=CLIENT_ID&scope=SCOPE&client_secret=CLIENT_SECRET";
+        given(applicationParams.getGrantType()).willReturn("GRANT_TYPE");
+        given(applicationParams.getClientId()).willReturn("CLIENT_ID");
+        given(applicationParams.getScope()).willReturn("SCOPE");
+        given(applicationParams.getClientSecret()).willReturn("CLIENT_SECRET");
+        given(activeDirectoryApiClient.authenticate(requestString)).willReturn(new AuthenticationResponse());
+
+    }
+
+    @Test
+    void shouldInitiateRequestHearing() {
```

Comment:
All these tests we should be able to `verify` that the correct method from the client was called correctly at least? And error conditions

Reply:
done